geode-core/src/test/java/org/apache/geode/management/internal/beans/stats/VMStatsMonitorTest.java

```diff
@@ -0,0 +1,106 @@
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import org.apache.geode.management.internal.MBeanJMXAdapter;
+import org.apache.geode.test.junit.categories.StatisticsTest;
+
+@Category(StatisticsTest.class)
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore("*.UnitTest")
+@PrepareForTest(MBeanJMXAdapter.class)
+public class VMStatsMonitorTest {
+  @Rule
+  public TestName testName = new TestName();
+
+  private VMStatsMonitor vmStatsMonitor;
+
+  @Before
+  public void setUp() {
+    PowerMockito.mockStatic(MBeanJMXAdapter.class);
+  }
+
+  @Test
+  public void statisticInitialValueShouldBeZeroWhenTheProcessCpuTimeJmxAttributeIsAvailable() {
+    when(MBeanJMXAdapter.isAttributeAvailable(anyString(), anyString())).thenReturn(true);
+    vmStatsMonitor = new VMStatsMonitor(testName.getMethodName());
+    assertThat(vmStatsMonitor).isNotNull();
+    assertThat(vmStatsMonitor.getCpuUsage()).isEqualTo(0);
+  }
+
+  @Test
+  public void statisticInitialValueShouldBeUndefinedWhenTheProcessCpuTimeJmxAttributeIsNotAvailable() {
+    when(MBeanJMXAdapter.isAttributeAvailable(anyString(), anyString())).thenReturn(false);
+    vmStatsMonitor = new VMStatsMonitor(testName.getMethodName());
+    assertThat(vmStatsMonitor).isNotNull();
+    assertThat(vmStatsMonitor.getCpuUsage()).isEqualTo(VMStatsMonitor.VALUE_NOT_AVAILABLE);
+  }
+
+  @Test
+  public void refreshStatsShouldUpdateCpuUsage() {
+    ZonedDateTime now = ZonedDateTime.now();
+    when(MBeanJMXAdapter.isAttributeAvailable(anyString(), anyString())).thenReturn(true);
+    vmStatsMonitor = spy(new VMStatsMonitor(testName.getMethodName()));
+    assertThat(vmStatsMonitor).isNotNull();
+    assertThat(vmStatsMonitor.getCpuUsage()).isEqualTo(0);
+    Number processCpuTime = spy(Number.class);
+    vmStatsMonitor.statsMap.put(StatsKey.VM_PROCESS_CPU_TIME, processCpuTime);
+
+    // First Run: updates lastSystemTime
+    when(vmStatsMonitor.currentTimeMillis()).thenReturn(now.toInstant().toEpochMilli());
+    vmStatsMonitor.refreshStats();
+    assertThat(vmStatsMonitor.getCpuUsage()).isEqualTo(0);
+    verify(processCpuTime, times(0)).longValue();
+
+    // Second Run: updates lastProcessCpuTime
+    when(processCpuTime.longValue()).thenReturn(500L);
+    vmStatsMonitor.refreshStats();
+    assertThat(vmStatsMonitor.getCpuUsage()).isEqualTo(0);
+    verify(processCpuTime, times(1)).longValue();
+
+    // Next runs will update the actual cpuUsage
+    for (int i = 2; i < 6; i++) {
+      long mockProcessCpuTime = i * 500;
+      long mockSystemTime = now.plus(i, ChronoUnit.SECONDS).toInstant().toEpochMilli();
+      when(processCpuTime.longValue()).thenReturn(mockProcessCpuTime);
+      when(vmStatsMonitor.currentTimeMillis()).thenReturn(mockSystemTime);
+
+      vmStatsMonitor.refreshStats();
+      verify(vmStatsMonitor, times(1)).calculateCpuUsage(mockSystemTime, mockProcessCpuTime);
+      assertThat(vmStatsMonitor.getCpuUsage()).isNotEqualTo(0);
```

Comment:
Currently this is returning `NaN`. Try adding:

```
      assertThat(cpuUsage).isNotEqualTo(Float.NaN);
      assertThat(cpuUsage).isLessThan(1F);
```

This will at least give us _some_ coverage of `calculateCpuUsage`. A dedicated test would be an improvement.

Reply:
Changes done.